Source/Core/DolphinQt/QtUtils/SignalBlocking.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <QSignalBlocker>
+
+// Helper class for populating a GUI element without triggering its data change signals.
+
+template <typename T>
+class SignalBlockerProxy
+{
+public:
+  explicit SignalBlockerProxy(T* object) : m_object(object), m_blocker(object) {}
+  SignalBlockerProxy(const SignalBlockerProxy& other) = delete;
```

Comment:
if the copy variants are deleted, you may want to also explicitly default or delete the move equivalents as well.